### PR TITLE
feat: new modules and February updates

### PR DESCRIPTION
**Pull request by Tomas2D**
- update `text` / `tune` schema
- add `tag` / `folder` and `text experimental` module (with `rerank`, `sentence-similarity`, `classification`)
- rename some schemas and add a fallback import support
- regenerate test cassettes
- update tests
- add `input` to the `text generation` service
- add new feedback service (under `request` module) and deprecated old `text.generation.feedback` module
- remove `max_new_tokens` upper limit

Ref: #332, #333, #334

### CODE REVIEW

File: examples/extensions/localserver/local_server.py [hunk 1]
```diff
@@ -83,7 +83,8 @@ def tokenize(self, input_text: str, parameters: TextTokenizationParameters) -> T
 print(heading("Use a local server with a custom model"))
 
 # Instantiate the Local Server with your model
-server = LocalLLMServer(models=[FlanT5Model])
+# set interface="0.0.0.0" to listen on all interfaces
+server = LocalLLMServer(models=[FlanT5Model], interface="127.0.0.1")
```
Comment (Tomas2D) [inline examples/extensions/localserver/local_server.py:87]: I've seen some examples where binding to `0.0.0.0` silently failed.

**Review by jezekra1 — CHANGES_REQUESTED** [review]

File: examples/tag/tag.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+client = Client(credentials=Credentials.from_env())
+
+print(heading("Show all existing tags"))
+for tag in client.tag.list(
+    limit=100,  # optional
+    offset=0,  # optional
+    type=TagType.LANGUAGE,  # optional
+).results:
+    pprint(tag.model_dump())
```
[the remark this answers is not in the document]
Comment (Tomas2D) [reply examples/tag/tag.py:32]: There is not any.

File: src/genai/schema/_api.py [hunk 3]
```diff
@@ -296,22 +373,23 @@ class TextGenerationComparisonCreateResultsParameters(ApiBaseModel):
...
 
 class TextGenerationFeedbackCategory(str, Enum):
-    NO_PROBLEM = "no_problem"
-    HAP = "hap"
-    PII = "pii"
-    SOCIAL_BIAS = "social_bias"
-    NOT_HONEST_OR_TRUTHFUL = "not_honest_or_truthful"
-    TABOO_TOPICS = "taboo_topics"
-    OTHER = "other"
+    INACCURATE = "inaccurate"
```
[the remark this answers is not in the document]
Comment (Tomas2D) [reply src/genai/schema/_api.py:371]: This one needs to be revised.

The feedback endpoint has been introduced in version `2023-11-22` and then updated in version `2024-02-20` (enum update), and now it has been deprecated in favour of `/requests/:id/feedback`.

I would update it and add the note to the changelog when releasing. What do you think?

File: examples/task/task.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    return "\n" + f" {text} ".center(80, "=") + "\n"
+
+
+client = Client(credentials=Credentials.from_env())
+
+print(heading("Show all existing tasks"))
+for task in client.task.list(
+    tune=True,  # optional
+).results:
+    pprint(task.model_dump())
```
[the remark this answers is not in the document]
Comment (Tomas2D) [reply examples/task/task.py:29]: It's obvious from the output. Now it's just tuning.

**Review by jezekra1 — APPROVED** [review]

File: src/genai/schema/_api.py [hunk 5]
```diff
@@ -242,6 +251,14 @@ class RequestApiVersion(str, Enum):
     V2 = "v2"
 
 
+class RequestChatConversationIdRetrieveResultsRequest(ApiBaseModel):
```
Comment (jezekra1) [inline src/genai/schema/_api.py:254]: should we name it `AnyDictionary` and unify it as before?
(would require extra aliases I believe)

Comment (Tomas2D) [reply src/genai/schema/_api.py:254]: Why? We just remove it in the next major version.

File: tests/integration/user/cassettes/test_user_service/TestUserService.test_update_user.yaml [hunk 6]
```diff
@@ -16,29 +16,28 @@ interactions:
     uri: https://api.com/v2/user?version=2023-11-22
   response:
     body:
-      string: '{"result":{"id":26916,"first_name":"Radek","last_name":"Jezek","tou_accepted":false,"data_usage_consent":true}}'
+      string: '{"result":{"id":453,"first_name":"Tomas","last_name":"Dvorak","email":"tomas.dvorak@ibm.com","tou_accepted":false,"data_usage_consent":true}}'
```
Comment (jezekra1) [inline tests/integration/user/cassettes/test_user_service/TestUserService.test_update_user.yaml:19]: funny that our email is baked in in the tests :D

**Review by jezekra1 — APPROVED** [review]

**Pull request merged**